sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedParquetRecordReader.java

```diff
@@ -324,6 +331,10 @@ public boolean nextBatch() throws IOException {
       }
       cv.assemble();
     }
+    // If needed, compute row indexes within a file.
+    if (rowIndexGenerator != null) {
+      rowIndexGenerator.populateRowIndex(columnVectors, num);
```

Comment:
Sorry I'm not very familiar with parquet read codebase, does this happen after row group skipping or not?

Reply:
Was looking through the code and it seems like it happens after row group skipping. Also, the tests in `ParquetRowIndexSuite` check that the rowIndex column doesn't have any of the values from the skipped row groups.